### PR TITLE
Softmax cross entropy ignore

### LINKED ISSUE #427: Support ignore label in `softmax_cross_entropy`
I want to ignore parts of labels in `softmax_cross_entropy`, for example output labels of RNNs.

### PR DESCRIPTION
**Pull request by unnonouno**
fix #427

### CODE REVIEW

File: chainer/functions/loss/softmax_cross_entropy.py [hunk 1]
```diff
@@ -34,14 +36,15 @@ def forward_cpu(self, inputs):
...
         # deal with the case where the SoftmaxCrossEntropy is
         # unpickled from the old version
         if getattr(self, 'normalize', True):
-            count = x.size // x.shape[1]
+            count = (t != self.ignore_label).sum()
         else:
             count = x.shape[0]
-        y = numpy.log(p).sum(keepdims=True) * (-1.0 / count)
+        y = (numpy.log(p) * (t.flat != self.ignore_label)).sum(keepdims=True) \
+            * (-1.0 / count)
```
Comment (okuta) [inline chainer/functions/loss/softmax_cross_entropy.py]: What happens when `count` equals 0?

Comment (unnonouno) [inline chainer/functions/loss/softmax_cross_entropy.py]: It's a good question!

**Comment by okuta** [comment]
LGTM!
